### PR TITLE
partitions/authmethod-index work from enterprise

### PR DESCRIPTION
**Pull request by markan**
Signed-off-by: Mark Anderson <manderson@hashicorp.com>

### CODE REVIEW

**Review by rboyer — APPROVED** [review]
LGTM

File: agent/consul/state/acl_schema.go [hunk 1]
```diff
@@ -499,3 +482,35 @@ func indexExpiresFromACLToken(raw interface{}, local bool) ([]byte, error) {
...
+		Name: tableACLAuthMethods,
+		Indexes: map[string]*memdb.IndexSchema{
+			indexID: {
+				Name:         indexID,
+				AllowMissing: false,
+				Unique:       true,
+				Indexer: &memdb.StringFieldIndex{
+					Field:     "Name",
+					Lowercase: true,
+				},
```
Comment (rboyer) [inline agent/consul/state/acl_schema.go:497]: This is missing the `indexerSingle` stuff.

**Review by rboyer — CHANGES_REQUESTED** [review]
The schema in OSS is missing something.

**Review by dhiaayachi — APPROVED** [review]
LGTM!

**Review by rboyer — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/454662.

**Pull request merged**